platform_plugin_elm_credentials/api/views.py

```diff
@@ -120,38 +130,152 @@ def get(self, request, course_id: str) -> HttpResponse:
                 status_code=status.HTTP_400_BAD_REQUEST,
             )
 
-        credential_username = query_params.get("username")
+        if username := query_params.get("username"):
+            json_data = self.generate_single_credential(
+                username, course_id, course_block, query_params
+            )
+
+            if isinstance(json_data, HttpResponse):
+                return json_data
+
+            if query_params.get("to_file"):
+                response = self.to_file(
+                    f"credential-{username}-{course_id}.json",
+                    json_data,
+                    "application/json",
+                )
+                return response
```

Comment:
I don't think we need the `to_file` queryparam option anymore, since the client already specified that the zip option was the only one used so that can be the default (see comment in jira for more context). We would have less points of failure by removing unused code, what do you think?

Reply:
I agree, I will remove the query param.